.github/workflows/build.yaml

```diff
@@ -0,0 +1,122 @@
+name: Build & Test
+
+on: [ push ]
+jobs:
+  tests:
+    name: Build & Test on ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ ubuntu-latest ]
+        go-version: [ 1.17.x ]
+    steps:
+      - name: Install Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: ${{ matrix.go-version }}
+
+      - name: Checkout code
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 1
+
+      - name: Get dependencies
+        run: go get -v -t -d ./...
+
+      - name: Build project
+        run: go build ./...
+
+      - name: Run tests
+        run: go test -race ./...
+
+      - id: results
+        shell: bash
+        env:
+          FULL_REVISION: ${{ github.event.after }}
+        run: |
+          echo "::set-output name=rev::${FULL_REVISION:0:8}"
+          echo "::set-output name=branch::${GITHUB_REF#refs/heads/}"
+          echo "::set-output name=os::$(uname -sr)"
+          echo "::set-output name=go_version::$(go version | cut -d " " -f 3-)"
+
+          if [[ "${{ job.status }}" == "success" ]]; then
+            echo "::set-output name=status:::moneybag: SUCCESS"
+          else
+            echo "::set-output name=status:::hankey: FAILURE"
+          fi
+
+      - name: Post to Slack
+        id: slack
+        uses: slackapi/slack-github-action@v1.16.0
+        with:
+          # Slack channel id, channel name, or user id to post message.
+          # See also: https://api.slack.com/methods/chat.postMessage#channels
+          channel-id: 'CCFC9EWH2'
```

Comment:
Can this be made a secret as well?  Mainly, I don't want to publish things like this in code if I can help it.  It is a soft secret.

Reply:
Done. The channel ID is useless without the token, but I agree.